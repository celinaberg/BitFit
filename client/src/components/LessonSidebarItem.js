// @flow

import React, { Component } from "react";
import { NavItem, NavLink } from "reactstrap";
import { NavLink as Link } from "react-router-dom";

class LessonSidebarItem extends Component {
  props: {
    id: string,
    title: string,
    admin: boolean
  };

  static defaultProps = {
    admin: false
  };

  render() {
    let url = "/lessons/" + this.props.id;
    if (this.props.admin) {
      url = "/admin/lessons/" + this.props.id;
    }

    return (
      <NavItem key={this.props.id}>
        <NavLink tag={Link} to={url}>
          {this.props.title}
        </NavLink>
      </NavItem>
    );
  }
}

export default LessonSidebarItem;
