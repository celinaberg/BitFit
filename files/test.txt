14 5.6 8.7 9.3
8
9.0
4.5
7
8.9 1.6 2.6
