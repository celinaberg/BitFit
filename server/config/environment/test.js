'use strict';

// Test specific configuration
// ===========================
module.exports = {
  // MongoDB connection options
  mongo: {
    uri: 'mongodb://localhost/its110-test'
  }
};