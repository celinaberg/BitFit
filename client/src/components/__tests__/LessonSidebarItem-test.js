// @flow

import React from "react";
import LessonSidebarItem from "../LessonSidebarItem";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import test from "ava";

test.skip("URL changes when admin", t => {
  let component = renderer.create(
    <MemoryRouter>
      <LessonSidebarItem id="testId" title="Test Lesson" />
    </MemoryRouter>
  );

  let tree = component.toJSON();
  t.snapshot(tree);

  component = renderer.create(
    <MemoryRouter>
      <LessonSidebarItem id="testId" title="Test Lesson" admin={true} />
    </MemoryRouter>
  );
  tree = component.toJSON();
  t.snapshot(tree);
});
