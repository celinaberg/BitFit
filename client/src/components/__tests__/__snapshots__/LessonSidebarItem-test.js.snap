// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL changes when admin 1`] = `
<li
  className="nav-item"
>
  <a
<<<<<<< HEAD
=======
    aria-current={false}
>>>>>>> 76a1d40... Initial commit
    className="nav-link"
    href="/lessons/testId"
    onClick={[Function]}
    style={undefined}
  >
    Test Lesson
  </a>
</li>
`;

exports[`URL changes when admin 2`] = `
<li
  className="nav-item"
>
  <a
<<<<<<< HEAD
=======
    aria-current={false}
>>>>>>> 76a1d40... Initial commit
    className="nav-link"
    href="/admin/lessons/testId"
    onClick={[Function]}
    style={undefined}
  >
    Test Lesson
  </a>
</li>
`;
